### PR TITLE
Refactor Code For Easily Testing And Update First Test (Signup)

### PR DESCRIPTION
**Pull request by sieusaya149**
This is the PR for #6 
- Now we have separete db, mq and app by depency injection that lead to when testing we can easily to mocking and testing the real application 
- Setup global-setup, global-teardown for testing. The setup will start docker compose for 3 external services (mysql, rabbitmq and utils_test (service helping migrate database))
- Testcase used axious for client http and nock for isolation network for testing and also support is-ci for monitor the testing is in the ci or not (jenkin, github action).
- Currently, the basic mocking server is working fine, but need refractor in the future

### CODE REVIEW

File: db_manage.sh [hunk 1]
```diff
@@ -9,35 +12,46 @@ display_usage() {
...
+        if [[ "$*" == *"-t"* ]] || [[ "$*" == *"--test"* ]]; then
+          CONTAINER_NAME="utils_test"
+        fi
+      fi
+      case "$key" in
+        --setup | -s)
+          docker exec "$CONTAINER_NAME" db_manager --setup
+          ;;
+        --cleanup | -c)
+          docker exec -it "$CONTAINER_NAME" db_manager --cleanup
```
Comment (hikijs) [inline db_manage.sh:33]: now with the -it option might be we can not run the script automatically

File: server.js [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+const {initializeWebServer} = require('./src/app')
+// Using Dependence Injection For DB and Rabbit Mq for easily testing
+async function start() {
+        console.time(">>> Start WebServer")
+        await initializeDatabase()
+        try {
+          await initializeWebServer();
+          await initializeRabbitMQ(RabbitMq)
+        } catch (error) {
+          releaseDatabaseConnection()
```
Comment (hikijs) [inline server.js:14]: I think should move the logic releaseDatabaseConnection to after the catch at line 24, **and check if already has the connection or not if not release for more better.**

File: src/dbs/init.mysql.js [hunk 3]
```diff
@@ -36,21 +36,31 @@ class Database
         return Database.instance
     }
 
+	releaseAllConnection()
+	{
+		if(this.typeConnection == TYPE_CONNECTION_DB.pool)
+		{
+			this.pool.end()
+		}
```
Comment (hikijs) [inline src/dbs/init.mysql.js:44]: should check the pool is null or not before end()

```suggestion
		if(this.typeConnection == TYPE_CONNECTION_DB.pool && this.pool)
		{
			this.pool.end()
		}
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: db_manage.sh (hunk 1 above)
Comment (sieusaya149) [reply db_manage.sh:33]: Oke I think we will update this in refactor commit

File: server.js (hunk 2 above)
Comment (sieusaya149) [reply server.js:14]: Oke I understand I will change it in the next commit

File: src/dbs/init.mysql.js (hunk 3 above)
Comment (sieusaya149) [reply src/dbs/init.mysql.js:44]: Thanks I will update in the next commit

**Comment by sieusaya149** [comment]
I will update request change in the next commit

**Pull request closed without merging**